backend/src/main/java/com/github/stackscrubs/stuq/backend/service/SessionService.java

```diff
@@ -0,0 +1,39 @@
+package com.github.stackscrubs.stuq.backend.service;
+
+import com.github.stackscrubs.stuq.backend.model.UserCredentials;
+import com.github.stackscrubs.stuq.backend.model.InvalidCredentialsException;
+import com.github.stackscrubs.stuq.backend.model.SessionNotFoundException;
+import com.github.stackscrubs.stuq.backend.model.jpa.Session;
+import com.github.stackscrubs.stuq.backend.model.jpa.User;
+import com.github.stackscrubs.stuq.backend.repository.SessionRepository;
+import com.github.stackscrubs.stuq.backend.repository.UserRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SessionService {
```

Comment:
Move logger from endpoint to here, as the majority of logic finds place in this file.

Reply:
Then you could do e.g:
```
User user = userRepository.findByEmail(credentials.getEmail())
            .filter((x) -> x.passwordMatches(credentials.getPassword()))
            .orElseThrow(() -> {
                    logger.info("someinfo");
                    return new InvalidCredentialsException()
            });
```